godtools/GeneralExtensions/UIFont.swift

```diff
@@ -11,30 +11,61 @@ import UIKit
 extension UIFont {
     
     static func gtRegular(size: CGFloat) -> UIFont {
-        return UIFont.systemFont(ofSize: size, weight: UIFontWeightRegular).transformToAppropriateFont()
+        return UIFont.buildGTFont(size: size, weight: UIFontWeightRegular)
     }
     
     static func gtLight(size: CGFloat) -> UIFont {
-        return UIFont.systemFont(ofSize: size, weight: UIFontWeightLight).transformToAppropriateFont()
+        return UIFont.buildGTFont(size: size, weight: UIFontWeightLight)
     }
     
     static func gtThin(size: CGFloat) -> UIFont {
-        return UIFont.systemFont(ofSize: size, weight: UIFontWeightThin).transformToAppropriateFont()
+        return UIFont.buildGTFont(size: size, weight: UIFontWeightThin)
     }
     
     static func gtSemiBold(size: CGFloat) -> UIFont {
-        return UIFont.systemFont(ofSize: size, weight: UIFontWeightSemibold).transformToAppropriateFont()
+        return UIFont.buildGTFont(size: size, weight: UIFontWeightSemibold)
     }
     
-    func transformToAppropriateFont() -> UIFont {
-        if GTSettings.shared.primaryLanguageId == nil {
-            return self
+    static func buildGTFont(size: CGFloat, weight: CGFloat) -> UIFont {
+        if shouldTransformLanguage() {
+            return UIFont.buildCustomGTFontWithSize(size, weight: weight)
+        } else {
+            return UIFont.systemFont(ofSize: size, weight: weight)
+        }
+    }
+    
+    static func buildCustomGTFontWithSize(_ size: CGFloat, weight: CGFloat) -> UIFont {
+        guard let language = LanguagesManager.defaultLanguage else {
+            return UIFont.systemFont(ofSize: size, weight: weight)
         }
         
-        let languagesManager = LanguagesManager()
-        let language = languagesManager.loadPrimaryLanguageFromDisk()
+        var fontName = ""
         
-        return transformToAppropriateFontByLanguage(language!, textScale: 1.0)
+        if language.code == "am-ET" {
```

Comment:
Let's do `if language.code != "am"` as the primary condition. Most of the time it will be true this way and we can just return the system font as normal.